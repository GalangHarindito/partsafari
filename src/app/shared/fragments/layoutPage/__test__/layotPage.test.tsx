import { render } from '@testing-library/react';
import LayoutPage from '../layoutPage';
import '@testing-library/jest-dom';

describe('LayoutPage Component', () => {
  test('Render the header with correct title', () => {
    const { container, getByText } = render(
      <LayoutPage>
        <div>Test Children</div>
      </LayoutPage>
    );
    const title = getByText(/to do/i);

    expect(title).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  test('Render children', () => {
    const { getByText } = render(
      <LayoutPage>
        <div>Test Children</div>
      </LayoutPage>
    );

    const childElement = getByText(/test children/i);
    expect(childElement).toBeInTheDocument();
  });
});
