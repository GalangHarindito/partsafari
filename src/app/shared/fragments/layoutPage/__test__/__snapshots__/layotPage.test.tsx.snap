// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutPage Component Render the header with correct title 1`] = `
<div>
  <section
    class="bg-background h-screen w-full"
  >
    <header
      class="bg-header-blue py-2.5 px-5 rounded-t-round-30px"
    >
      <h2
        class="text-white font-bold  xl:text-3xl lg:text-2xl md:text-xl"
      >
        To Do
      </h2>
    </header>
    <main
      class="bg-background-main h-screen w-full rounded-b-round-30px pt-12.5 flex justify-center"
    >
      <div>
        Test Children
      </div>
    </main>
  </section>
</div>
`;
